File: candy.py
class Solution:
    # @param ratings, a list of integer
    # @return an integer
    def candy(self, ratings):
    	if len(ratings) == 1:
    		return 1
    	count = [1 for i in range(len(ratings))]
    	for r in range(len(ratings) - 1):
    		if ratings[r] < ratings[r+1] and count[r]>=count[r+1]:
    			count[r+1] = count[r]+1
    		#print count
    	#print "#"
    	for r in range(len(ratings) - 1, 0, -1):
    		if ratings[r] < ratings[r-1] and count[r-1]<=count[r]:
    			count[r-1] = count[r]+1
    		#print count
    	return sum(count)


s = Solution()
print s.candy([1,3,5])

File: JAVA/candy.java
public class Solution {
    public int candy(int[] ratings) {
        if (ratings.length <= 1)
            return ratings.length;
        int[] kids = new int[ratings.length];
        int count = 0;
        for (int j = 0; j < ratings.length; j++)
            kids[j] = 1;
        for (int j = 1; j < ratings.length; j++)
            if (ratings[j] > ratings[j-1] && kids[j] <= kids[j-1])
                kids[j] = kids[j-1]+1;
        
        for (int j = ratings.length - 2; j >= 0 ; j--)
            if (ratings[j] > ratings[j+1] && kids[j] <= kids[j+1])
                kids[j] = kids[j+1]+1;
        
        for (int j = 0; j < ratings.length; j++)
            count += kids[j];
        return count;
    }
}